internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -1465,20 +1465,21 @@ func flattenKubernetesClusterDataSourceMicrosoftDefender(input *managedclusters.
 }
 
 func flattenKubernetesClusterDataSourceUpgradeSettings(input *managedclusters.AgentPoolUpgradeSettings) []interface{} {
-	maxSurge := ""
+	values := make(map[string]interface{})
+
 	if input != nil && input.MaxSurge != nil {
-		maxSurge = *input.MaxSurge
+		values["max_surge"] = *input.MaxSurge
 	}
 
-	if maxSurge == "" {
-		return []interface{}{}
+	if input != nil && input.DrainTimeoutInMinutes != nil {
+		values["drain_timeout_in_minutes"] = *input.DrainTimeoutInMinutes
 	}
 
-	return []interface{}{
-		map[string]interface{}{
-			"max_surge": maxSurge,
-		},
+	if input != nil && input.DrainTimeoutInMinutes != nil {
+		values["node_soak_duration_in_minutes"] = *input.NodeSoakDurationInMinutes
 	}
+
+	return []interface{}{values}
```

Comment:
We usually nil check the input first and return early if it's empty, GH won't allow me to use the suggestions feature on this snippet but could we update this to 

```
	if input == nil {
		return []interface{}{}
	}

	values := make(map[string]interface{})
	
	if input.MaxSurge != nil {
		values["max_surge"] = *input.MaxSurge
	}

	if input.DrainTimeoutInMinutes != nil {
		values["drain_timeout_in_minutes"] = *input.DrainTimeoutInMinutes
	}

	if input.DrainTimeoutInMinutes != nil {
		values["node_soak_duration_in_minutes"] = *input.NodeSoakDurationInMinutes
	}

	return []interface{}{values}
```